### PR TITLE
fix support for newer git (doesn't work with...

### PR DESCRIPTION
**Pull request by nickboldt**
fix support for newer git (doesn't work with --remote=path/on/local; use pushd and simpler git archive command so we can test this script locally);
fix support for dotnet31 devfile, which doesn't use master branch;
tweak console output and arg desc (cache_project REQUIRES branch, it's not optional)

Change-Id: Ia7ffb1791b04e83456be9e1a978b69ee1116fb36
Signed-off-by: nickboldt <nboldt@redhat.com>

### CODE REVIEW

**Comment by nickboldt** [comment]
See more discussion/context/explanation in https://github.com/redhat-developer/codeready-workspaces/pull/247 re: https://issues.redhat.com/browse/CRW-644

**Comment by nickboldt** [comment]
when approved, should also cherry pick to 7.9.x

File: build/scripts/cache_projects.sh [hunk 1]
```diff
@@ -24,14 +24,17 @@ TEMP_REPO="${TEMP_DIR}/cloned"
...
 function cache_project() {
   local repo="$1"
   local branch="$2"
   local destination="$3"
-  git clone "$repo" -b "$branch" --depth 1 "$TEMP_REPO" &>/dev/null
-  git archive "$branch" --remote="$TEMP_REPO" --format zip --output "$destination"
+  rm -fr "$TEMP_REPO"
+  git clone "$repo" -b "$branch" --depth 1 "$TEMP_REPO" -q
+  pushd "$TEMP_REPO" &>/dev/null
+    git archive "$branch" -o "$destination"
```
Comment (JPinkney) [inline build/scripts/cache_projects.sh]: Should this command be in line with the other ones?

Comment (nickboldt) [reply build/scripts/cache_projects.sh]: Could be, but that block is "pushd" in. ;)

(Personal style choice. Helps me remember when I'm in a pushd/popd block.)

Comment (JPinkney) [reply build/scripts/cache_projects.sh]: Ah gotcha!, that makes sense!

**Pull request merged**